### ISSUE TITLE
PM2 heredoc support for bash

### ISSUE THREAD

**Issue by abale** [Unitech/pm2#1848]
pm2's json app configuration is great, however it does not comply with bash heredoc standard, meaning you have to have a physical .json file to load it into pm2.

Here is what you could achieve (think from an automated deployment point of view) if the input was heredoc compliant:

```
pm2 start <<EOF
{
"apps" : [{
    "name"        : "eventTrack",
    "script"      : "index.js",
    "args"        : [],
    "watch"       : true,
    "instances"   : 4,
    "node_args"   : "",
    "merge_logs"  : true,
    "cwd"         : "$1/myapppath/",
    "env": {
        "NODE_ENV": "production",
        "MONGOCOL":"log",
        "REDIS_PORT":6379
    }
  }]
}
EOF
```

Now if my $1 parameter has my working path (which in automation/containerized environment can be dynamic and hard to meaningfully predict), I can start pm2 on that instance quite easily, without having to write a json file to disk and read it for this exclusive reason.

**Comment by joaosa** [comment]
Yes, this would be quite convenient to have in automated deployment scenarios.
